src/poller/utils.rs

```diff
@@ -106,6 +126,50 @@ fn any_annotation(json: &JsonValue) -> Option<Vec<String>> {
     return None;
 }
 
+/// Returns true if a editor is marked spammer
+pub async fn get_is_editor_spammer(
+    editor_id: i32,
+    pool: &PgPool
+) -> bool {
+    let query = r#"
+    SELECT privs & 4096 != 0 as is_editor_spammer
+    FROM editor
+    WHERE id = $1;
+    "#;
+    let (is_editor_spammer, ) = sqlx::query_as::<_, (bool, )>(query)
+        .bind(editor_id)
+        .fetch_one(pool)
+        .await
+        .unwrap();
+    return is_editor_spammer
+}
+
+/// Returns the edit type if the editor is not spammer, `None` the editor is spammer
+pub async fn get_edit_type_if_editor_is_not_spammer(
+    edit_id: i32,
+    pool: &PgPool
+) -> Option<i16> {
+    let query = r#"
+    SELECT edit.type
+    FROM edit
+    JOIN editor ON edit.editor = editor.id
+    WHERE edit.id = $1
+    AND (editor.privs & 4096) = 0;
+    "#;
+
+    let res = sqlx::query_as::<_, (i16, )>(query)
+        .bind(edit_id)
+        .fetch_one(pool)
+        .await;
+    match res {
+        Ok((id, )) => { Some(id) }
+        Err(e) => {
+            eprintln!("error: {}", e);
```

Comment:
Instead of printing and returning none, it might be better to actually handle the error case outside of the function. 
But might be better for another PR

Reply:
Cool I'll return a Result wrapped later then